File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/PassThroughCesiumUriResolver.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 *  
 A URI resolver that leaves URIs unchanged.
 

 */
@SuppressWarnings("unused")
public class PassThroughCesiumUriResolver implements ICesiumUriResolver {
    /**
    *  
    Resolves a URI, leaving it unchanged.
    
    
    

    * @param uri The source URI.
    * @return The same URI.
    */
    public final String resolveUri(String uri) {
        return uri;
    }

    /**
    *  Gets a static instance of {@link PassThroughCesiumUriResolver} usable from any thread.
    

    */
    @Nonnull
    public static PassThroughCesiumUriResolver getInstance() {
        return s_instance;
    }

    @Nonnull
    private static PassThroughCesiumUriResolver s_instance = new PassThroughCesiumUriResolver();
}

File: DotNet/CesiumLanguageWriter/Advanced/PassThroughCesiumUriResolver.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A URI resolver that leaves URIs unchanged.
    /// </summary>
    public class PassThroughCesiumUriResolver : ICesiumUriResolver
    {
        /// <summary>
        /// Resolves a URI, leaving it unchanged.
        /// </summary>
        /// <param name="uri">The source URI.</param>
        /// <returns>The same URI.</returns>
        public string ResolveUri(string uri)
        {
            return uri;
        }

        /// <summary>
        /// Gets a static instance of <see cref="PassThroughCesiumUriResolver"/> usable from any thread.
        /// </summary>
        [NotNull]
        public static PassThroughCesiumUriResolver Instance
        {
            get { return s_instance; }
        }

        [NotNull]
        private static readonly PassThroughCesiumUriResolver s_instance = new PassThroughCesiumUriResolver();
    }
}